### DISCUSSION TITLE
Migrating from verdaccio instance to another verdaccio instance

### DISCUSSION THREAD

**Discussion by mbareck7** [verdaccio/verdaccio#3003] [Q&A]
This is directly related to the issue  [1318](https://github.com/verdaccio/verdaccio/issues/1318)

**What am I doing:** 
copying the data folder from on verdaccio instance to another verdaccio instance

### Current install (on perm server, ubuntu 18.04):

```
    ~$ verdaccio --version
    v5.1.2
    
    ~$ verdaccio --info
    Environment Info:
    System:
        OS: Linux 5.4 Ubuntu 18.04.6 LTS (Bionic Beaver)
        CPU: (20) x64 Intel(R) Xeon(R) Silver 4114 CPU @ 2.20GHz
    Binaries:
        npm: 7.20.3 - ~/.nvm/versions/node/v16.7.0/bin/npm
    Virtualization:
        Docker: 20.10.7 - /usr/bin/docker
    Browsers:
        Firefox: 97.0
```

verdaccio config :

```storage: /home/admin/verdaccio/storage
plugins: /home/admin/verdaccio/plugins

web:
  title: Verdaccio

auth:
  htpasswd:
    file: /home/admin/verdaccio/htpasswd

uplinks:
  npmjs:
    url: https://registry.npmjs.org/

packages:
  '@*/*':
    access: $all
    publish: $authenticated
    unpublish: $authenticated
    proxy: npmjs

  '**':

    access: $all

    publish: $authenticated
    unpublish: $authenticated

    proxy: npmjs

server:
  keepAliveTimeout: 300

middlewares:
  audit:
    enabled: true

logs:
  - { type: stdout, format: pretty, level: http }

listen: 0.0.0.0:4873
```

### New install (on kubernetes cluster ): 
installation guide : https://verdaccio.org/docs/kubernetes/

Usefull info from the deployment file
```
        app: npm-verdaccio
        app.kubernetes.io/instance: npm
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: verdaccio
        app.kubernetes.io/version: 5.5.0
        helm.sh/chart: verdaccio-4.6.1

        image: verdaccio/verdaccio:5.2.0
```

verdaccio config (the ConfigMap ): 

```---
# Source: verdaccio/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: npm-verdaccio
  namespace: verdaccio
  labels:
    helm.sh/chart: verdaccio-4.6.1
    app.kubernetes.io/name: verdaccio
    app.kubernetes.io/instance: npm
    app.kubernetes.io/version: "5.5.0"
    app.kubernetes.io/managed-by: Helm
    app: npm-verdaccio
data:
  config.yaml: |-
    
    # path to a directory with all packages
    storage: /verdaccio/storage/data
    
    web:
      # WebUI is enabled as default, if you want disable it, just uncomment this line
      #enable: false
      title: Verdaccio
    
    auth:
      htpasswd:
        # Do not change this path if secrets htpasswd is used.
        file: /verdaccio/storage/htpasswd
        max_users: -1
    
    # a list of other known repositories we can talk to
    uplinks:
      npmjs:
        url: https://registry.npmjs.org/
        agent_options:
          keepAlive: true
          maxSockets: 40
          maxFreeSockets: 10
    
    packages:
      '@*/*':
        # scoped packages
        access: $all
        publish: $authenticated
        proxy: npmjs
    
      '**':
        access: $all
        publish: $authenticated
    
        proxy: npmjs
    
    middlewares:
      audit:
        enabled: true
    
    logs: {type: stdout, format: pretty, level: http}
```

### Performed actions : 
copy one project from /home/admin/verdaccio/storage on the current install to /verdaccio/storage/data on the new install, using 

``` kubectl cp my-lib npm-verdaccio-pod:/verdaccio/storage/data/  ```

ssh into the pod as root then
``` chmod -R 777 /verdaccio/storage ```

### After  a restart 
**Expected results, seeing the newly transferred package**
**Actual result : No Package Published Yet.**

**Comment by mbareck7** [comment]
@juanpicado  Okay, let's discuss here !

**Reply by juanpicado** [reply]
Thanks for the reminder, I'll put it on my list. But if the topic is Kubernetes, I'd ask https://github.com/verdaccio/charts here there are way smarter than me.

**Comment by sacru2red** [comment]
It may have already been resolved, but in my case, [bulk-npm-publish](https://www.npmjs.com/package/bulk-npm-publish) was helpful.

**Reply by mbareck7** [reply]
Thank you @sacru2red for sharing, the problem is solved, in fact the libraries are accessible but they are not displayed on the UI.

**Reply by kotori** [reply]
@mbareck7 How were you able to get your libraries to finally display on the UI?

**Reply by mbtools** [reply]
The [Verdaccio database](https://verdaccio.org/docs/configuration#.verdaccio-db) must include the package names.

PS: This is only true, if you use the default storage. If using a storage plugin, then check with the plugin owner.
